pygmt/tests/test_grdimage.py

```diff
@@ -117,6 +117,22 @@ def test_grdimage_shading_xarray(grid, shading):
     return fig_ref, fig_test
 
 
+@check_figures_equal()
+def test_grdimage_grid_and_shading_with_xarray(grid, xrgrid):
+    """
+    Test that shading works well when xarray.DataArray is input to both the
+    ``grid`` and ``shading`` arguments.
+    """
+    fig_ref, fig_test = Figure(), Figure()
+    fig_ref.grdimage(
+        grid="@earth_relief_01d_g", region="GL", cmap="geo", shading=xrgrid, verbose="i"
+    )
+    fig_ref.colorbar()
+    fig_test.grdimage(grid=grid, region="GL", cmap="geo", shading=xrgrid, verbose="i")
+    fig_test.colorbar()
```

Comment:
Btw, I tried using `makecpt` to get both plots to use the same colormap scale, but the images still differ, appears to be some slight offset in the left-right direction.

```suggestion
    makecpt(cmap="geo", series=(-3900, 3200))
    fig_ref.grdimage(
        grid="@earth_relief_01d_g", region="GL", cmap=True, shading=xrgrid, verbose="d"
    )
    fig_ref.colorbar()
    fig_test.grdimage(grid=grid, region="GL", cmap=True, shading=xrgrid, verbose="d")
    fig_test.colorbar()
```

| Expected (correct) | Generated (wrong) | Diff |
|--|--|--|
| ![test_grdimage_grid_and_shading_with_xarray png -expected](https://user-images.githubusercontent.com/23487320/121279988-9b395500-c929-11eb-8a75-74fdf97219a6.png) | ![test_grdimage_grid_and_shading_with_xarray png](https://user-images.githubusercontent.com/23487320/121279987-9a082800-c929-11eb-881a-c73c4f64168a.png) | ![test_grdimage_grid_and_shading_with_xarray png -failed-diff](https://user-images.githubusercontent.com/23487320/121279990-9b395500-c929-11eb-94d8-6960d449e846.png) |